source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,2515 @@
+    `GridFS <../source/gridfs/gridfs-spec.rst#configurable-gridfsbucket-class>`__
+    specification. The ``readConcern``, ``readPreference``, and ``writeConcern``
+    options use the same structure as defined in `Common Options`_.
+
+.. _entity_stream:
+
+- ``stream``: Optional document. Corresponds with a stream as defined in the
+  `GridFS <../gridfs/gridfs-spec.rst>`__ spec. Test runners MUST ensure that
+  stream is both readable *and* writable.
+
+  This entity is primarily used with `uploadFromStream`_ and
+  `uploadFromStreamWithId`_ operations for `bucket`_ entities.
+
+  The structure of this document is as follows:
+
+  - ``id``: Required string. Unique name for this entity. The YAML file SHOULD
+    define a `node anchor`_ for this field (e.g. ``id: &stream0 stream0``).
+
+  - ``hexBytes``: Required string. The string MUST contain an even number of
+    hexademical characters (case-insensitive) and MAY be empty. The test runner
+    MUST raise an error if the string is malformed. The test runner MUST convert
+    the string to a byte sequence denoting the stream's readable data (if any).
+    For example, "12ab" would denote a stream with two bytes: "0x12" and "0xab".
+
+
+collectionData
+~~~~~~~~~~~~~~
+
+List of documents that should correspond to the contents of a collection. This
+structure is used by both `initialData`_ and `test.outcome <test_outcome_>`_,
+which insert and read documents, respectively.
+
+The structure of this document is as follows:
+
+- ``collectionName``: Required string. See `commonOptions_collectionName`_.
+
+- ``databaseName``: Required string. See `commonOptions_databaseName`_.
+
+- ``documents``: Required array of documents. List of documents corresponding to
+  the contents of the collection. This list may be empty.
+
+
+test
+~~~~
+
+Test case consisting of a sequence of operations to be executed.
+
+The structure of each document is as follows:
+
+- ``description``: Required string. The name of the test.
+
+  This SHOULD describe the purpose of this test (e.g. "insertOne is retried").
+
+.. _test_runOn:
+
+- ``runOn``: Optional array of documents. List of server version and/or topology
+  requirements for which the tests in this file can be run. If specified, these
+  requirements are evaluated independently and in addition to any top-level
+  `runOn`_ requirements. If no requirements in this array are met, the test
+  runner MUST skip this test.
+
+  These requirements SHOULD be more restrictive than those specified in the
+  top-level `runOn`_ requirements (if any). They SHOULD NOT be more permissive.
+  This is advised because both sets of requirements MUST be satisified in order
+  for a test to be executed and more permissive requirements at the test-level
+  could be taken out of context on their own.
+
+  If set, the array should contain at least one document. The structure of each
+  document is defined in `runOnRequirement`_.
+
+.. _test_skipReason:
```

Comment:
I see many parallels between `test` and the top-level `spec`,  proposal to add `skipReason` to the top level?